packages/components/src/components/wizard/wizard.jsx

```diff
@@ -1,86 +1,99 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
-import Header from './wizard-header.jsx';
-import Icon from '../icon/icon.jsx';
-
-const Wizard = React.forwardRef(({ children, steps, lbl_previous, has_prev, lbl_next, has_next, className }, ref) => {
-    const [step, setStep] = React.useState(0);
-    const [component, setComponent] = React.useState(steps[0].component);
-    const last_step = steps.length - 1;
-
-    const nextStep = () => {
-        const next_step = step + 1;
-        const is_last_step = next_step > last_step;
-        if (is_last_step) return;
-        setStep(next_step);
-        setComponent(steps[next_step].component);
-    };
+import Step from './wizard-step.jsx';
+
+const Wizard = ({ children, className, initial_step, onStepChange, nav }) => {
+    const [active_step, setActiveStep] = React.useState(0);
+
+    React.useEffect(() => {
+        const local_initial_step = initial_step - 1;
+        const local_children = getSteps();
 
-    const prevStep = () => {
-        const prev_step = step - 1;
-        const is_first_step = prev_step < 0;
-        if (is_first_step) return;
-        setStep(prev_step);
-        setComponent(steps[prev_step].component);
+        if (local_initial_step && local_children[local_initial_step]) {
+            setActiveStep(initial_step);
+        }
+    }, [initial_step, getSteps]);
+
+    const onChangeStep = stats => {
+        // User callback
+        onStepChange(stats);
     };
 
-    const selectStep = selected_step => {
-        const is_prev_step_not_allowed = selected_step < 0;
-        const is_next_step_not_allowed = selected_step > last_step;
-        if (is_prev_step_not_allowed || is_next_step_not_allowed) return;
-        setStep(selected_step);
-        setComponent(steps[selected_step].component);
+    const isInvalidStep = next => next < 0 || next >= totalSteps();
+
+    const onSetActiveStep = next => {
+        if (active_step === next) return;
+        if (isInvalidStep(next)) return;
+
+        setActiveStep(next);
+        onChangeStep({
+            active_step: next + 1,
+        });
     };
 
-    React.useImperativeHandle(ref, () => ({
-        prevStep,
+    const getSteps = React.useCallback(() => React.Children.toArray(children), [children]);
+
+    const currentStep = () => active_step + 1;
+
+    const totalSteps = () => getSteps().length;
+
+    const goToStep = step => onSetActiveStep(step - 1);
+
+    const firstStep = () => goToStep(1);
+
+    const lastStep = () => goToStep(totalSteps());
+
+    const nextStep = () => onSetActiveStep(active_step + 1);
+
+    const previousStep = () => onSetActiveStep(active_step - 1);
```

Comment:
I think using verbs here really helps to distinguish methods.

Reply:
Yes I will update all the names.